float64[7] pq
